packages/web/src/components/background-animations/MusicConfetti.tsx

```diff
@@ -84,65 +37,41 @@ type MusicConfettiProps = {
   isMatrix?: boolean
 }
 
-const defaultProps: MusicConfettiProps = {
-  withBackground: false,
-  recycle: false,
-  limit: 250,
-  friction: 0.99,
-  gravity: 0.2,
-  rotate: 0.1,
-  swing: 0.01,
-  particleRate: 0.1,
-  zIndex: 16,
-  onCompletion: () => {},
-  isMatrix: false
-}
-
-const MusicConfetti = (props: MusicConfettiProps) => {
-  const newProps = { ...props } as Required<MusicConfettiProps>
-
-  if (props.isMatrix) {
-    newProps.swing = 0
-    newProps.rotate = 0
-  }
-
-  return <UnconditionalMusicConfetti {...newProps} />
-}
-
-MusicConfetti.defaultProps = defaultProps
-
-const UnconditionalMusicConfetti = (props: Required<MusicConfettiProps>) => {
-  const confettiPromiseRef = useRef<Promise<Confetti | undefined> | null>(null)
-
-  // When we mount canvas, start confetti and set the confettiPromiseRef
-  const {
-    recycle,
-    limit,
-    friction,
-    gravity,
-    rotate,
-    swing,
-    particleRate,
-    onCompletion,
-    isMatrix
-  } = props
-
+export const MusicConfetti = ({
+  recycle,
+  limit,
+  friction,
+  gravity,
+  rotate,
+  swing,
+  particleRate,
+  zIndex,
+  withBackground,
+  onCompletion,
+  isMatrix
+}: MusicConfettiProps) => {
+  const confettiRef = useRef<Confetti | null>(null)
+
+  // When we mount canvas, start confetti and set the confettiRef
   const setCanvasRef = useCallback(
     (node: HTMLCanvasElement) => {
       if (!node) return
-      const confetti = startConfettiAnimation(
+
+      const confetti = new Confetti(
         node,
-        recycle,
-        limit,
-        friction,
-        gravity,
-        rotate,
-        swing,
-        particleRate,
-        onCompletion,
-        isMatrix
+        isMatrix ? undefined : PATHS,
+        isMatrix ? undefined : COLORS,
+        recycle || false,
+        limit || 200,
+        friction || 0.99,
+        gravity || 0.2,
+        isMatrix ? 0 : rotate || 0.1,
+        isMatrix ? 0 : swing || 0.01,
+        particleRate || 0.1,
```

Comment:
Will also advocate for magic numbers to move to constants at the top if they have semantic meaning and might need to be tweakable.